app/javascript/components/Schedule.js

```diff
@@ -0,0 +1,197 @@
+      .flat()
+      .find(slot => slot.id === targetSlot.id);
+    let previousSessionID = slot.program_session_id;
+
+    slot.program_session_id = programSession.id;
+
+    if (programSession.slot) {
+      this.unscheduleSession(programSession);
+    }
+
+    if (previousSessionID && previousSessionID !== slot.program_session_id) {
+      let replacedSession = this.state.sessions.find(
+        session => session.id === previousSessionID
+      );
+      unscheduledSessions.push(replacedSession);
+    }
+
+    this.setState({ unscheduledSessions, schedule });
+  };
+
+  unscheduleSession = programSession => {
+    let unscheduledSessions = this.state.unscheduledSessions.slice();
+    let previousSlot = programSession.slot;
+
+    let schedule = Object.assign({}, this.state.schedule);
+    let day = previousSlot.conference_day.toString();
+
+    let slot = Object.values(schedule.slots[day])
+      .flat()
+      .find(slot => slot.id === previousSlot.id);
+
+    slot.program_session_id = null;
+    unscheduledSessions.push(Object.assign(programSession, { slot: null }));
+
+    this.setState({ unscheduledSessions, schedule });
+  };
+
+  componentDidMount() {
+    let hours = this.determineHours(this.props.schedule.slots);
+    const trackColors = palette("tol-rainbow", this.props.tracks.length);
+    this.props.tracks.forEach((track, i) => {
+      track.color = "#" + trackColors[i];
+    });
+
+    this.setState(Object.assign(this.state, this.props, hours)); // doing this until I can hit the API here.
+  }
+
+  componentDidUpdate() {
+    console.log(this.state);
+  }
+
+  render() {
+    const {
+      counts,
+      dayViewing,
+      startTime,
+      endTime,
+      schedule,
+      sessions,
+      unscheduledSessions,
+      draggedSession,
+      csrf,
+      tracks
+    } = this.state;
+
+    const headers = schedule.rooms.map(room => (
+      <div className="schedule_column_head" key={'column_head_' + room.name}>
+        {room.name}
+      </div>
+    ));
+
+    const headersMinWidth = (180 * schedule.rooms.length) + 'px'
+
+    return (
+      <div className="schedule_grid">
+        <Nav
+          counts={counts}
+          changeDayView={this.changeDayView}
+          dayViewing={dayViewing}
+          schedule={schedule}
+        />
+        <div className="grid_headers_wrapper" style={{'minWidth': headersMinWidth}}>{headers}</div>
+        <div className="grid_container">
+          <Ruler startTime={startTime} endTime={endTime} />
+          <DayView
+            schedule={schedule}
+            dayViewing={dayViewing}
+            startTime={startTime}
+            endTime={endTime}
+            ripTime={this.ripTime}
+            changeDragged={this.changeDragged}
+            draggedSession={draggedSession}
+            csrf={csrf}
+            sessions={sessions}
+            scheduleSession={this.scheduleSession}
+            tracks={tracks}
+          />
+          <UnschedledArea
```

Comment:
`UnschedledArea` ->`UnscheduledArea` (missing 'u')

Reply:
I'm impressed this didn't cause me issues. Good eye.